packages/block-editor/src/hooks/auto-inserting-blocks.js

```diff
@@ -95,13 +89,21 @@ function AutoInsertingBlocksControl( props ) {
 						if ( autoInsertedBlock ) {
 							clientIds[ block.name ] =
 								autoInsertedBlock.clientId;
+						} else {
+							// If no auto-inserted block was found in any of its designated locations,
+							// we check if it's present elsewhere in the block tree.
+							// If it is, we consider it manually inserted and remove the corresponding
+							// toggle from the block inspector panel.
+							const blockIsPresentElsewhereInTree = someBlock(
+								getBlocks(),
```

Comment:
Have you tried using `getGlobalBlockCount`?

https://github.com/WordPress/gutenberg/blob/4cd832016762791b1bfb2651cca148a0d0642996/packages/block-editor/src/store/selectors.js#L301L313

I don’t know reliable it is with controlled inner blocks like Template Part or Reusable Block. It operates on client ids so maybe it’s fine.

Reply:
Oh, thanks a lot for the pointer! I hadn't seen that one.

It seems to be working fine! It finds the Mini Cart block when I move it elsewhere (even inside a template part) 🎉